include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -79,6 +79,18 @@ class all_view
         cgh.require(m_acc);
     }
 
+  private:
+    accessor_t create_accessor(sycl::buffer<_T, 1> __buf, diff_type __offset, diff_type __n)
```

Comment:
it is dangerous to call non-static function in initialization. As I see, you don't need `this`, so please make it static.